packages/ui/src/components/IconResolver/IconResolver.tsx

```diff
@@ -20,10 +20,12 @@ export const IconResolver: FunctionComponent<PropsWithChildren<IconResolverProps
       );
     case CatalogKind.Processor:
     case CatalogKind.Component:
+      // eslint-disable-next-line no-case-declarations
+      const iconType = props.tile.type === CatalogKind.Processor ? NodeIconType.EIP : NodeIconType.Component;
       return (
         <img
           className={props.className}
-          src={NodeIconResolver.getIcon(props.tile.name)}
+          src={NodeIconResolver.getIcon(props.tile.name, iconType)}
```

Comment:
If we successfully determine all icons usage, we could pass the `CatalogKind` type directly to the method. Alternatively, if for some reason knowing the `processorName` + `componentName` is useful, we could change the API to pass that instead.